### DISCUSSION TITLE
How to create global mutable variables in Typst?

### DISCUSSION THREAD

**Discussion by PEWOLO** [typst/typst#3309] [Q&A] [answered]
Hey everyone,

I was wondering if there is a way to create global variables  whose values can be changed from  in a function in Typst. 
Any idea would be super helpful.

**Comment by Dherse** [comment, marked as answer]
Have a look at [`state`](https://typst.app/docs/reference/introspection/state/). It is the only solution as otherwise, global variables would break memoization and purity.

**Reply by PEWOLO** [reply]
Thanks for the link!
That handles what I wanted to do

**Reply by Dherse** [reply]
No worries, have fun using typst
